components/builder-depot/src/server.rs

```diff
@@ -1251,187 +955,90 @@ fn dont_cache_response(response: &mut Response) {
+        get "/views/:channel/pkgs/:origin/:pkg/latest" => show_package,
+        get "/views/:channel/pkgs/:origin/:pkg/:version" => list_packages,
+        get "/views/:channel/pkgs/:origin/:pkg/:version/latest" => show_package,
+        get "/views/:channel/pkgs/:origin/:pkg/:version/:release" => show_package,
         post "/views/:channel/pkgs/:origin/:pkg/:version/:release/promote" => {
-            move |r: &mut Request| promote_package(&depot8, r)
-        },
-        get "/views/:view/pkgs/:origin/:pkg/:version/:release/download" => {
-            move |r: &mut Request| download_package(&depot40, r)
-        },
-        get "/views/:view/origins/:origin/keys" => move |r: &mut Request| list_origin_keys(&depot41, r),
-        get "/views/:view/origins/:origin/keys/latest" => {
-            move |r: &mut Request| download_latest_origin_key(&depot42, r)
-        },
-        get "/views/:view/origins/:origin/keys/:revision" => {
-            move |r: &mut Request| download_origin_key(&depot43, r)
-        },
-
-        get "/pkgs/search/:query" => move |r: &mut Request| search_packages(&depot9, r),
-        get "/pkgs/:origin" => move |r: &mut Request| list_packages(&depot10, r),
-        get "/pkgs/:origin/:pkg" => move |r: &mut Request| list_packages(&depot11, r),
-        get "/pkgs/:origin/:pkg/latest" => move |r: &mut Request| show_package(&depot12, r),
-        get "/pkgs/:origin/:pkg/:version" => move |r: &mut Request| list_packages(&depot13, r),
-        get "/pkgs/:origin/:pkg/:version/latest" => {
-            move |r: &mut Request| show_package(&depot14, r)
-        },
-        get "/pkgs/:origin/:pkg/:version/:release" => {
-            move |r: &mut Request| show_package(&depot15, r)
-        },
-
-        get "/pkgs/:origin/:pkg/:version/:release/download" => {
-            move |r: &mut Request| download_package(&depot16, r)
+            XHandler::new(promote_package).before(basic)
         },
+        get "/views/:view/pkgs/:origin/:pkg/:version/:release/download" => download_package,
+        get "/views/:view/origins/:origin/keys" => list_origin_keys,
+        get "/views/:view/origins/:origin/keys/latest" => download_latest_origin_key,
+        get "/views/:view/origins/:origin/keys/:revision" => download_origin_key,
+
+        get "/pkgs/search/:query" => search_packages,
+        get "/pkgs/:origin" => list_packages,
+        get "/pkgs/:origin/:pkg" => list_packages,
+        get "/pkgs/:origin/:pkg/latest" => show_package,
+        get "/pkgs/:origin/:pkg/:version" => list_packages,
+        get "/pkgs/:origin/:pkg/:version/latest" => show_package,
+        get "/pkgs/:origin/:pkg/:version/:release" => show_package,
+
+        get "/pkgs/:origin/:pkg/:version/:release/download" => download_package,
         post "/pkgs/:origin/:pkg/:version/:release" => {
-            move |r: &mut Request| upload_package(&depot17, r)
+            if depot.config.insecure {
+                XHandler::new(upload_package)
```

Comment:
Just tried it out (uploading to a local `hab-depot start --insecure`), works like cake!